libtsuba/src/RDGSlice.cpp

```diff
@@ -15,29 +15,33 @@ tsuba::RDGSlice::DoMake(
   ReadGroup grp;
   katana::Uri topology_path =
       metadata_dir.Join(core_->part_header().topology_path());
-  katana::Uri node_types_path =
-      metadata_dir.Join(core_->part_header().node_entity_type_id_array_path());
-  katana::Uri edge_types_path =
-      metadata_dir.Join(core_->part_header().edge_entity_type_id_array_path());
-
   KATANA_CHECKED_CONTEXT(
       core_->topology_file_storage().Bind(
           topology_path.string(), slice.topo_off,
           slice.topo_off + slice.topo_size, true),
       "loading topology array");
-  KATANA_CHECKED_CONTEXT(
-      core_->node_entity_type_id_array_file_storage().Bind(
-          node_types_path.string(),
-          slice.node_range.first * sizeof(katana::EntityTypeID),
-          slice.node_range.second * sizeof(katana::EntityTypeID), true),
-      "loading node type id array");
-  KATANA_CHECKED_CONTEXT(
-      core_->edge_entity_type_id_array_file_storage().Bind(
-          edge_types_path.string(),
-          slice.edge_range.first * sizeof(katana::EntityTypeID),
-          slice.edge_range.second * sizeof(katana::EntityTypeID), true),
-      "loading edge type id array");
 
+  if (core_->part_header().IsEntityTypeIDsOutsideProperties()) {
+    katana::Uri node_types_path = metadata_dir.Join(
+        core_->part_header().node_entity_type_id_array_path());
+    katana::Uri edge_types_path = metadata_dir.Join(
+        core_->part_header().edge_entity_type_id_array_path());
+
+    KATANA_CHECKED_CONTEXT(
+        core_->node_entity_type_id_array_file_storage().Bind(
+            node_types_path.string(),
+            slice.node_range.first * sizeof(katana::EntityTypeID),
```

Comment:
Having to do this math is laborious and error prone. Consider adding a TODO where this interface can be improved to handle data type sizes automatically.

Reply:
yes. My point being that having to do this math is error prone. It's like calling `malloc` vs `new` and people forget to multiply length with `sizeof(T)`.

Reply:
I did some work on `FileView` about a year ago and my thoughts then and now are that we should offer a few flavors of `FileView`. It currently serves several purposes and it doesn't make sense to add special functionality for each of them. For example, it doesn't make sense to have a `Bind()` variant that converts number_of_elements -> number_of_bytes for an object that is used to read Parquet files. Just like it doesn't make sense to have prefetching for an object that is used to read slices of binary arrays.

But if we had a `ParquetView`, a `BinaryArrayView`, and a `CSRView` that each maintained a `FileView` internally (via subclassing or pointers), we could have good interfaces for all of those use cases and put only truly common functionality in `FileView`.

Reply:
I see. I'll leave it up to you to decide how to proceed. I won't be touching `FileView` in the near future (not yet).